File: easy/3005_count_elements_with_maximum_frequency/solution.py
from collections import Counter
from typing import List


class Solution:
    def maxFrequencyElements(self, nums: List[int]) -> int:
        nums_counter = Counter(nums)
        max_frequency = max(nums_counter.values())
        return max_frequency * sum(
            1
            for frequency in nums_counter.values()
            if frequency == max_frequency
        )

File: easy/3005_count_elements_with_maximum_frequency/solution.cpp
using namespace std;

#include <unordered_map>
#include <vector>

class Solution {
public:
  int maxFrequencyElements(vector<int> &nums) {
    unordered_map<int, int> numsCounter;
    numsCounter.reserve(nums.size());

    int maxFrequency = 0;
    int maxFrequencyCount = 0;
    int numCount;

    for (int &num : nums) {
      numCount = ++numsCounter[num];

      if (numCount > maxFrequency) {
        maxFrequency = numsCounter[num];
        maxFrequencyCount = 1;
      } else if (numCount == maxFrequency) {
        maxFrequencyCount++;
      }
    }

    return maxFrequency * maxFrequencyCount;
  }
};
